packages/block-library/src/navigation-link/index.php

```diff
@@ -177,7 +177,22 @@ function render_block_core_navigation_link( $attributes, $content, $block ) {
 	// Don't render the block's subtree if it is a draft or if the ID does not exist.
 	if ( $is_post_type && $navigation_link_has_id ) {
 		$post = get_post( $attributes['id'] );
-		if ( ! $post || 'publish' !== $post->post_status ) {
+		/**
+		 * Filter allowed post_status for navigation link block to render.
+		 *
+		 * @since 6.7.0
```

Comment:
```suggestion
		 * @since 6.8.0
```

This will be released in WordPress 6.8 so well need to adjust this.